File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentence(string text);


int main(void)
{
    string text = get_string("Text: ");
    float letters = count_letters(text);
    float word = count_words(text);
    float sentence = count_sentence(text);
    // apply the Coleman-Liau index
    float grade = (0.0588 * (letters / word) * 100) - (0.296 * (sentence / word) * 100) - 15.8;
    // round grade
    grade = round(grade);
    if (grade >= 1 && grade <= 16)
    {
        printf("Grade %.f\n", grade);
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

// function to count the letters in a string
int count_letters(string text)
{
    int letters = 0;
    int l = strlen(text);
    for (int i = 0; i < l; i++)
    {
        // only count alphabtes
        if isalpha(text[i])
        {
            letters++;
        }
    }
    return letters;

}

// function to count the words in a string
int count_words(string text)
{
    int word = 1;
    int l = strlen(text);
    for (int i = 0; i < l; i++)
    {
        // count after every space
        if isspace(text[i])
        {
            word++;
        }
    }
    return word;
}


// function to count sentences in a string
int count_sentence(string text)
{
    int sentence = 0;
    int l = strlen(text);
    for (int i = 0; i < l; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
    return sentence;
}

File: sentimental-readability/readability.py
# TODO

letters = sentences = 0
words = 1


def main():
    text = input("Text: ")
    counter(text)

    # apply the Coleman-Liau index

    grade = round((0.0588 * (letters / words) * 100) - (0.296 * (sentences / words) * 100) - 15.8)

    # print correct response

    if grade >= 1 and grade <= 16:
        print(f"Grade {grade}")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print("Grade 16+")


def counter(text):
    global letters, words, sentences
    for char in text.lower():  # loop through input
        if char.isalpha():  # increment letters if alpha
            letters += 1
        elif char.isspace():  # increment words if space
            words += 1
        elif char in [".", "!", "?"]:  # increment sentences if .?!
            sentences += 1


if __name__ == "__main__":
    main()